sessionctx/variable/varsutil.go

```diff
@@ -427,9 +427,13 @@ func ValidateSetSystemVar(vars *SessionVars, name string, value string) (string,
 		if !exists {
 			return "", ErrWrongValueForVar.GenWithStackByArgs(name, value)
 		}
-		switch upVal {
-		case "SERIALIZABLE", "READ-UNCOMMITTED":
-			return "", ErrUnsupportedValueForVar.GenWithStackByArgs(name, value)
+
+		strictCompatibility, err := GetSessionSystemVar(vars, TiDBStrictCompatibility)
+		if !TiDBOptOn(strictCompatibility) || err != nil {
+			switch upVal {
+			case "SERIALIZABLE", "READ-UNCOMMITTED":
+				return "", ErrUnsupportedValueForVar.GenWithStackByArgs(name, value)
+			}
```

Comment:
Maybe we should throw a warning here.

Reply:
@winkyao see https://github.com/pingcap/tidb/issues/2712 for context. Users could lose data from incorrect assumptions, it is better not to throw a warning because it can be unsafe to ignore.

Reply:
@morgo I guess @winkyao means report a warning here even the flag is on.

Reply:
@zhexuany OK! That makes sense to me too. It does not match the name `skip-*` though unfortunately since it is still checking but no longer generating an error.

Reply:
It is better to append the warning, e.g.
https://github.com/pingcap/tidb/pull/10065/files#diff-3ed884d9f6758ab4a4a44e0a884d4909R290